RHDEVDOCS-2815 Authenticating pipelines using git secret (#37610)
openshift/openshift-docs: modules/op-understanding-credential-selection.adoc

> // This module is included in the following assembly:

Reviewer: Readers LOVE examples like the ones you've provided here! In web analytics for technical documentation, "examples" is one of the most frequently used search terms.

//
// *openshift-docs/cicd/pipelines/authenticating-pipelines-using-git-secret.adoc

[id="op-understanding-credential-selection_{context}"]

> = Understanding credential selection

Reviewer: IBM SG: "If the subject is a functional overview, begin a heading with words such as Introduction or Overview rather than contriving a pseudo-task-oriented heading that begins with Understanding, Using, Introducing, or Learning."
```suggestion
= Credential selection
```

> A pipeline run or task run may require multiple authentications for accessing different Git repositories. You must annotate each secret to specify the domains for which {pipelines-shortname} can use the credential contained in a  secret.

Reviewer: ```suggestion
A pipeline run or task run might require multiple authentications to access different Git repositories. Annotate each secret with the domains where {pipelines-shortname} can use its credentials.
```

Reviewer: For **May**, the IBM SG says `Use [may] only to indicate permission. Otherwise, use "can" to indicate ability, or use "might" to indicate possibility.`

Reviewer: About "for accessing" in the previous sentence, do you find that the "to <verb>" form of the phrase is simpler and easier to read than "for <verb>ing"? These are small choices, but they add up to a style that can be easier or more difficult for readers to consume.

Reviewer: Should `domains` be singular?

A credential annotation key for Git secrets must begin with `tekton.dev/git-`, and its value is the URL of the host for which you want {pipelines-shortname} to use that credential.

> For example, {pipelines-shortname} uses a `basic-auth` secret (username and password) to access repositories at `github.com` and `gitlab.com`.

Reviewer: ```suggestion
In the following example, {pipelines-shortname} uses a `basic-auth` secret, which relies on a username and password, to access repositories at `github.com` and `gitlab.com`.
```
IBM SG, in the "Parentheses" sections, states "Use parentheses to identify items such as abbreviations, symbols, and measurements, but avoid using parentheses in general text."
Here, I've replaced the parentheses with a parenthetical phrase between commas. You could also remove that phrase altogether.

.Example: Multiple credentials for basic authentication
[source,yaml,subs="attributes+"]
----
apiVersion: v1
kind: Secret
metadata:
  annotations:
    tekton.dev/git-0: github.com
    tekton.dev/git-1: gitlab.com
type: kubernetes.io/basic-auth
stringData:
  username: <cleartext username>

>   password: <cleartext password or personal access token>

Reviewer: Should other examples also mention `cleartext` like this one?

----

> You can also use a `ssh-auth` secret (private key) to access a Git repository.

Reviewer: ```suggestion
You can also use an `ssh-auth` secret (private key) to access a Git repository.
```
Yes, it's strange, but choosing "a" or "an" depends on the _sound_ of the following consonant. Here, because SSH is pronounced "ess-ess-ayche," we use "an."
